### PR TITLE
SALTO-5522: Add data to trace logs of http calls

### PR DESCRIPTION
**Pull request by IdoZakk**
Data logs were missing in POST requests that were part of the deploy

---

I removed the existing data from the deploy log to avoid duplicate write

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/adapter-components/src/client/http_client.ts [hunk 1]
```diff
@@ -61,6 +61,8 @@ export type ClientDataParams = ClientBaseParams & {
 
 export type ClientParams = ClientBaseParams | ClientDataParams
 
+const isClientDataParams = (params: ClientParams): params is ClientDataParams => 'data' in params
```
Comment (shir-reifenberg) [inline packages/adapter-components/src/client/http_client.ts:64]: is there any difference from `isMethodWithData` ?

**Review by shir-reifenberg — APPROVED** [review]
LGTM 😎  with one question

File: packages/adapter-components/src/client/http_client.ts (hunk 1 above)
Comment (IdoZakk) [reply packages/adapter-components/src/client/http_client.ts:64]: No, missed that 🤭

File: packages/adapter-components/src/client/http_client.ts [hunk 2]
```diff
@@ -248,6 +249,7 @@ export abstract class AdapterHTTPClient<TCredentials, TRateLimitConfig extends C
           ? `<omitted buffer of length ${res.data.length}>`
           : this.clearValuesFromResponseData(res.data, url),
         headers: this.extractHeaders(res.headers),
+        data,
```
Comment (netama) [inline packages/adapter-components/src/client/http_client.ts:252]: please add a check that this is an object / array, if it's a buffer we should handle it like above

Comment (IdoZakk) [reply packages/adapter-components/src/client/http_client.ts]: I added a similar check to the above. Is this good, or do you prefer a more specific check?

Comment (IdoZakk) [reply packages/adapter-components/src/client/http_client.ts]: @netama I see you changed the file to be named deployment_old. Could not find the new place for deployment, should I make the change there as well?

Comment (netama) [reply packages/adapter-components/src/client/http_client.ts]: the logs are different in the new infra so no need, will try to pair them more before we upgrade the existing adapters but I don't think you have something specific to remove right now, thanks

**Pull request merged**